client/src/components/common/boost/BoostCalibration.tsx

```diff
@@ -0,0 +1,86 @@
+import React , {useCallback, useState} from 'react';
+import {Button, Card, InputGroup, FormControl} from 'react-bootstrap';
+import {setCalibration as sendCalibrationValue, resetCalibration} from 'api/common/powerModel';
+// import {useSensorData} from 'api/v2/sensors/data';
+
+/**
+ * Boost distance claibration input form
+ * 
+ * @returns {React.Component} Component
+ */
+function BoostCalibrationInput() {
+    const [calibrationValue, setCalib] = useState('');
+
+    const handleCalibrationChange = useCallback(
+        (event) => {
+        setCalib(event.target.value);
+        },
+        [setCalib],
+    );
+
+    const onSet = useCallback(
+        () => {
+          const calibValue: number = parseInt(calibrationValue, 10);
+          if (Number.isNaN(calibValue)) {
+            alert("Please enter an integer value only!");
+          }
+          else {
+            console.log(calibValue);
+            sendCalibrationValue(parseInt(calibrationValue, 10));
+          }
+          setCalib('');
+        },
+        [calibrationValue, setCalib],
+    );
+
+    const handleKeyPressed = useCallback(
+        (event) => { if (event.key === 'Enter') onSet(); },
+        [onSet]
+    );
+
+    return (
+    <InputGroup className="mb-1" >
+        <FormControl
+            onChange={handleCalibrationChange}
+            onKeyPress={handleKeyPressed}
+            placeholder="Calibrate distance..."
+            value={calibrationValue}
+        />
+        <InputGroup.Append className="l5">
+            <Button className="mr-2" variant="outline-secondary" onClick={onSet}>Set</Button>
+        </InputGroup.Append>
+        
+        <Button variant="outline-danger" onClick={resetCalibration}>Reset</Button>
+    </InputGroup>
+    );
+}
+
+/**
+ * Boost calibration card, displays the calibrated distance and the 
+ * distance travelled by the bike
+ * 
+ * @returns {React.Component} Component
+ */
+export default function BoostCalibration() {
+    // const GetDistTravelled = () => {
+    //     return useSensorData("reed_distance").data;
+    // };
+
+    return (
+        <Card >
+            <Card.Body>
+                <Card.Title>Calibration</Card.Title>
+                <Card.Text>
+                    BOOST may use a distance different to the bike&apos;s travelled distance for the purposes of generating power plan data.
+                </Card.Text>
+                <div className="pb-3">
+                    <b>Distance travelled (m) </b> <span className="float-right pr-4" > 30 </span>
```

Comment:
Can you pass the distance travelled/calibrated distance as a prop, rather than hardcode it here? You can of course still hardcode it in the View and the story. Let me know if you're unsure how to do this

Reply:
I think I have a general idea of how to go about it, but I was planning to do that in the next PR where I actually try to get the correct value. If you still want it done in this PR, let me know and I can attempt it.

Reply:
I'd rather it were done in this PR, because it allows us to see how this component will be used and verify that everything still displays properly when given different inputs, and that callbacks are called correctly.

Reply:
Do we specifically want it passed in from `BoostView` otherwise if the aim is to just use callbacks, I can pass it from `BoostCalibration` if I decide to keep the two classes separate.